packages/@lwc/babel-plugin-component/src/__tests__/wire-decorator.spec.js

```diff
@@ -97,14 +97,11 @@ describe('Transform property', () => {
                         return {
                           key2: ["fixed", "array"],
                           key1:
-                            host.prop1 != null
-                              ? host.prop1.prop2 != null
-                                ? host.prop1.prop2.prop3 != null
-                                  ? host.prop1.prop2.prop3.prop4 != null
-                                    ? host.prop1.prop2.prop3.prop4
-                                    : undefined
-                                  : undefined
-                                : undefined
+                            host.prop1 != null &&
```

Comment:
@jodarove @pmdartus @diervo any idea of how to optimize this? I'm thinking some memoization in between, something like:

```js
function config(host) {
    let key1 = host.prop1;
    // sub keys
    key1 = (key1 != null) && (key1 = key1.prop2) != null && (key1 = key1.prop3) != null && (key1 = key1.prop4) != null ? key1 : undefined;
    // return the config with each individual unique key
    return {
       key1,
   };
}
```

There are few invariants with this form:

1. if host.prop1 is a getter, or any other prop is a getter, it will be invoked only once.
2. since the props are unique on the config, the lets are unique as well.

This form should probably be easier as well, unless browsers are doing crazy optimizations here.